espnet2/diar/layers/tcn_nomask.py

```diff
@@ -0,0 +1,264 @@
+# Implementation of the TCN proposed in
```

Comment:
I thought they are similar to the code from `enh`, if so, we can just softlink to that instead of duplicating.

Reply:
Yes, it's based on the code from the `enh` task, but slightly different. It's split into `tcn_nomask.py` and `multi_mask.py` to select the masks corresponding to the number of speakers.

Reply:
We can do that, but I prefer not to touch the `enh` parts for now. This model structure is specific to this task and might be confusing for the contributors and the users of `enh` task if we make changes to `enh` codes.